pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -518,10 +520,15 @@ func (r *ReconcileBareMetalHost) actionProvisioning(prov provisioner.Provisioner
 		}
 		err = r.client.Get(context.TODO(), key, userDataSecret)
 		if err != nil {
-			return "", errors.Wrap(err,
+			return "", nil, errors.Wrap(err,
 				"failed to fetch user data from secret reference")
 		}
-		return string(userDataSecret.Data["userData"]), nil
+		var networkData map[string]interface{}
+		if err = yaml.Unmarshal(userDataSecret.Data["networkData"], &networkData); err != nil {
+			return "", nil, errors.Wrap(err, "failed to econde network_data.json from secret")
```

Comment:
typo s/econde/encode/

Also what happens in the case where networkData isn't defined in the secret - will this break?  Having a test to prove that would be nice :)

Reply:
Given the context, it should probably be "decode" or "unmarshal" rather than "encode".

Reply:
reading nonexistent key from map[string][]byte returns nil pointer to []byte, yaml.Unmarshal handles it properly so networkData will be nil as well which is ok for gophercloud module.

That being said this needs to be covered by test. Thanks for pointing this out :)